src/main/scala/is/hail/rvd/RVD.scala

```diff
@@ -401,6 +401,30 @@ trait RVD {
 
   def sample(withReplacement: Boolean, p: Double, seed: Long): RVD
 
+  def zipWithIndex(name: String): RVD
+
+  private[rvd] def zipWithIndexCRDD(
+    name: String
+  ): (TStruct, ContextRDD[RVDContext, RegionValue]) = {
+    val (newRowType, ins) = rowType.unsafeStructInsert(TInt64(), List(name))
+
+    val a = sparkContext.broadcast(countPerPartition().scanLeft(0L)(_ + _))
+
+    val newCRDD = crdd.cmapPartitionsWithIndex({ (i, ctx, it) =>
```

Comment:
don't we already know the partition counts sometimes? How can we incorporate that?

Reply:
Yeah we should definitely figure out how to incorporate that. I'd like to leave that to a later PR because Caitlin needs her pipeline to be not-slow before Friday and I've got a few other things to glue together before then.